serial/serialwin32.py

```diff
@@ -285,11 +285,20 @@ def read(self, size=1):
                     ctypes.byref(self._overlapped_read))
                 if not read_ok and win32.GetLastError() not in (win32.ERROR_SUCCESS, win32.ERROR_IO_PENDING):
                     raise SerialException("ReadFile failed ({!r})".format(ctypes.WinError()))
-                result_ok = win32.GetOverlappedResult(
-                    self._port_handle,
-                    ctypes.byref(self._overlapped_read),
-                    ctypes.byref(rc),
-                    True)
+                timeout = Timeout(self._timeout)
+                while True:
+                    # even with timeout infinite, return from kernel mode every 100ms to allow for things like KeyboardInterrupt to happen
+                    millis = win32.DWORD(int(min(timeout.time_left() or 0.1, 0.1) * 1000))
```

Comment:
if time_left() is 0, millis will be 0
if time_left() is < 0.1, millis will be time_left() * 1000
if time_left() is > 0.1, millis will be 0.1 * 1000 (capped)